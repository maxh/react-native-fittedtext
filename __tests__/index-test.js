import 'react-native';
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import {
  NativeModules
} from 'react-native';

import FittedText from '../index';

let savedRaf, savedMeasure;

const mockLayoutHelpers = (computeHeight) => {
  global.requestAnimationFrame = fn => fn();
  NativeModules.UIManager.measureLayoutRelativeToParent =
      (node, onError, onSuccess) => {
        onSuccess(0, 0, 0, computeHeight());
      };
};

describe('<FittedText />', () => {
  it('renders correctly with just targetHeight', () => {
    const tree = renderer.create(
      <FittedText targetHeight={100}>foo</FittedText>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with initialFontSize', () => {
    const tree = renderer.create(
      <FittedText targetHeight={100} initialFontSize={50}>foo</FittedText>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    NativeModules.UIManager.measureLayoutRelativeToParent = jest.fn();
  });

  it('finds a fitting fontSize, starting from a high initialFontSize', () => {
    const wrapper = shallow(
      <FittedText targetHeight={100}>foo</FittedText>);
    const instance = wrapper.instance();

    mockLayoutHelpers(() => instance.state.fontSize * 3);

    expect(instance.state.fontSize).toBe(100);
    return instance.componentDidMount().then(() => {
      expect(instance.state.fontSize).toBe(25);
    })
  });

  it('finds a fitting fontSize, starting from a low initialFontSize', () => {
    const instance = shallow(
      <FittedText targetHeight={100} initialFontSize={5}>
        foo
      </FittedText>).instance();

    mockLayoutHelpers(() => instance.state.fontSize * 3);

    expect(instance.state.fontSize).toBe(5);
    return instance.componentDidMount().then(() => {
      expect(instance.state.fontSize).toBe(28.125);
    })
  });

  beforeEach(() => {
    savedRaf = requestAnimationFrame;
    savedMeasure = NativeModules.UIManager.measureLayoutRelativeToParent;
  });

  afterEach(() => {
    requestAnimationFrame = savedRaf;
    NativeModules.UIManager.measureLayoutRelativeToParent = savedMeasure;
  });
});