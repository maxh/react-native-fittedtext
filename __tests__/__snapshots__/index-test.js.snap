exports[`<FittedText /> renders correctly with initialFontSize 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "backgroundColor": "transparent",
      "color": "transparent",
      "fontSize": 50,
    }
  }>
  foo
</Text>
`;

exports[`<FittedText /> renders correctly with just targetHeight 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "backgroundColor": "transparent",
      "color": "transparent",
      "fontSize": 100,
    }
  }>
  foo
</Text>
`;
